src/hotspot/share/classfile/classLoaderExt.cpp

```diff
@@ -336,7 +321,10 @@ ClassPathEntry* ClassLoaderExt::find_classpath_entry_from_cache(const char* path
   }
   ClassPathEntry* new_entry = NULL;
 
-  new_entry = create_class_path_entry(path, &st, false, false, false, CHECK_NULL);
+  ExceptionMark em(current);
+  Thread* THREAD = current; // For exception macros.
+  new_entry = create_class_path_entry(path, &st, /*throw_exception=*/false,
+                                      false, false, CATCH); // will never throw
```

Comment:
Given we are actually dealing with the VMThread which can't (or shouldn't!) actually throw exceptions I'm concerned about this chunk of code. I don't think we have a good idiom for these cases where the VMThread calls potentially exception throwing code in a way that should never throw an exception. Ideally all the exception-related fields would be in JavaThread.

Reply:
All code changed by this patch executes in the main Java thread.